.github/workflows/build.yaml

```diff
@@ -43,17 +47,12 @@ jobs:
           git fetch --prune --unshallow
       - name: Define version
         id: version
-        env:
-          hash: ${{ hashFiles('Dockerfile', '**/go.sum', '**/pkg/**', '**/examples/**', '**/cmd/**', '**/vendor/**', '**/Makefile', '**/main.go') }}
-        run: |
-          version="${{ env.hash }}"
-          version=${version::16}
-          echo "version=${version}" >> $GITHUB_OUTPUT
+        uses: ./.github/actions/version
       - name: Check cache for Toolkit image
         id: cache-toolkit
         uses: actions/cache/restore@v4
         env:
-          cache-name: pr-toolkit-build-x86_64
+          cache-name: toolkit-build-x86_64-${{ github.event_name }}
```

Comment:
This is used for PRs and for pushed to `main`, so to distinguish them we use the `github.event_name`, this way we ensure that once rebuilding in `main` after merging it does not reuse the cache from the PR that caused the push.